power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -172,6 +313,18 @@ template <dataset_type_tag dataset_type_> class Dataset {
         Idx const idx = find_component(ComponentType::name, false);
         return get_buffer_span_impl<StructType>(scenario, idx);
     }
+    template <template <class> class type_getter, class ComponentType,
+              class StructType = DataStruct<typename type_getter<ComponentType>::type>>
+    RangeObject<StructType> get_columnar_buffer_span(Idx scenario = invalid_index) const {
+        assert(scenario < batch_size());
```

Comment:
assert here if the buffer if the row buffer is `nullptr`.

Reply:
This check is done in `get_columnar_buffer_span_impl`